native-image-shared-config/pom.xml

```diff
@@ -99,14 +99,62 @@
         </plugin>
       </plugins>
     </pluginManagement>
-    <plugins>
-      <plugin>
-        <groupId>org.sonatype.plugins</groupId>
-        <artifactId>nexus-staging-maven-plugin</artifactId>
-      </plugin>
-    </plugins>
+    <extensions>
+      <extension>
+        <!--
+          Enables the "artifactregistry://" URL scheme (go/airlock/howto_maven).
+          Note that Maven extensions cannot be included in profiles (
+          https://maven.apache.org/guides/introduction/introduction-to-profiles.html#profiles-in-poms)
+        -->
+        <groupId>com.google.cloud.artifactregistry</groupId>
+        <artifactId>artifactregistry-maven-wagon</artifactId>
+        <version>2.2.3</version>
+      </extension>
+    </extensions>
   </build>
   <profiles>
+    <profile>
+      <!-- By default, we release artifacts to Sonatype, which requires
+          nexus-staging-maven-plugin. -->
+      <id>release-sonatype</id>
+      <activation>
+        <property>
+          <!-- Only when we use the release-gcp-artifact-registry profile,
+          which comes with artifact-registry-url property, this profile is
+          turned off. -->
+          <name>!artifact-registry-url</name>
```

Comment:
I don't think it's possible to chain profile activations/deactivations like this.

Reply:
This is not chaining profile activation (this is not relying on the line 145).

This activation relies on the fact that when we publish artifacts to GCP Artifact Registry using `release-gcp-artifact-registry` profile, `-Dartifact-registry-url=...` is always passed to the mvn command.

Reply:
I'll verify it works in the Kokoro release job.

Reply:
Why have line 145, then?

Reply:
Line 145 is dummy value that tells the property is undefined.

Reply:
> I'll verify it works in the Kokoro release job.

The release of this repository and google-auth-library-java (https://github.com/googleapis/google-auth-library-java/pull/1593) succeeded.